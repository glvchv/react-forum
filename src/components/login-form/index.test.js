import React from 'react';
import renderer from 'react-test-renderer';
import LoginForm from '.';
import { BrowserRouter } from 'react-router-dom';

jest.mock('../error-message', () => 'ErrorMessage');

describe('Login-form component', () => {
    it('should render form', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <LoginForm />
            </BrowserRouter>
        );
        expect(tree).toMatchSnapshot();
    });
})