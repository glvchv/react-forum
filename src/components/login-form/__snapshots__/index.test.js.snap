// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login-form component should render form 1`] = `
<div
  className="login-page"
>
  <div
    className="form"
  >
    <h1
      className="title"
    >
      Sign in
    </h1>
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <label
        className="label"
      >
        Username:
      </label>
      <input
        onChange={[Function]}
        placeholder="johndoe89"
        type="text"
      />
      <ErrorMessage
        error=""
      />
      <label
        className="label"
      >
        Password:
      </label>
      <input
        onChange={[Function]}
        placeholder="********"
        type="password"
      />
      <ErrorMessage
        error=""
      />
      <button
        disabled={false}
        type="submit"
      >
        login
      </button>
      <p
        className="message"
      >
        Not registered? 
        <a
          href="/register"
          onClick={[Function]}
        >
          Create an account
        </a>
      </p>
    </form>
  </div>
</div>
`;
